File: Book-AlgorithmsAndDataStructuresForProgrammingContests/cpp/065.cpp
// http://judge.u-aizu.ac.jp/onlinejudge/description.jsp?id=ALDS1_2_B&lang=ja
#include <iostream>
using namespace std;

int selectionSort(int A[], int N) {
  int minj, count;
  count = 0;
  for (int i = 0; i < N - 1; i++) {
    minj = i;
    for (int j = i; j < N; j++) {
      if (A[j] < A[minj]) minj = j;
    }
    swap(A[i], A[minj]);
    if (i != minj) count++;
  }
  return count;
}

int main() {
  int A[100], N, count;
  cin >> N;
  for (int i = 0; i < N; i++) cin >> A[i];
  count = selectionSort(A, N);

  for (int i = 0; i < N; i++) {
    if (i) cout << " ";
    cout << A[i];
  }
  cout << endl;
  cout << count << endl;
  return 0;
}

File: Book-AlgorithmsAndDataStructuresForProgrammingContests/julia/065.jl
struct RetStruct
  count
  retval
  RetStruct(count, retval) = new(count, retval)
end

function selectionsort(a)
  minj = 1
  count = 0
  N = length(a)
  for i in 1:N
    minj = i
    for j in i:N
      minj = a[j] < a[minj] ? j : minj
    end
    a[i], a[minj] = a[minj], a[i]
    count = i != minj ? count+1 : count
  end
  return RetStruct(count, a)
end

input = [5,6,4,2,1,3]
selectionsort(input)
